src/Raven.Server/Documents/Sharding/ShardedDocumentPutAction.cs

```diff
@@ -15,8 +16,11 @@ public ShardedDocumentPutAction(ShardedDocumentsStorage documentsStorage, Sharde
     }
 
     // TODO need to make sure we check that for counters/TS/etc...
-    public override void ValidateId(Slice lowerId)
+    public override void ValidateId(DocumentsOperationContext context, Slice lowerId, DocumentFlags documentFlags = DocumentFlags.None)
```

Comment:
what about deletion of a document?

Reply:
done in https://github.com/ravendb/ravendb/pull/17060